contrib/pkg/broker/user_provided/controller/controller.go

```diff
@@ -61,8 +72,13 @@ func (c *userProvidedController) Catalog() (*brokerapi.Catalog, error) {
 	}, nil
 }
 
-func (c *userProvidedController) CreateServiceInstance(id string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, error) {
+func (c *userProvidedController) CreateServiceInstance(
+	id string,
+	req *brokerapi.CreateServiceInstanceRequest,
+) (*brokerapi.CreateServiceInstanceResponse, error) {
 	credString, ok := req.Parameters["credentials"]
+	c.rwMutex.Lock()
+	defer c.rwMutex.Unlock()
```

Comment:
Not sure what the performance implications are of the marshal funcs but I would prefer if we only locked around the bits we're worried about and let the rest of things be mult-thread enabled.  IOW, would it break things if we moved this mutex to be around the update of the map and leave the rest of it outside of the mutex?  You could even move that update outside of the if-statement.